x-pack/plugins/alerting/README.md

```diff
@@ -0,0 +1,232 @@
+# Kibana alerting
+
+The Kibana alerting plugin provides a common place to setup alerts. It supports:
+
+- Registering types of alerts
+- List the registered types of alerts
+- CRUD on alerts
+
+## Terminology
+
+**Alert Type**: A function that takes parameters and fires actions to alert instances.
+
+**Alert**: A configuration that defines a schedule, an alert type w/ parameters, state information and actions.
+
+**Alert Instance**: The instances created from the an alert type execution.
+
+## Usage
+
+1. Create an alert type (see alert types -> example).
+2. Create an alert using the RESTful API (see alerts -> create).
+
+## Alert types
+
+### Methods
+
+**server.plugins.alerting.registerType(options)**
+
+The following table describes the properties of the `options` object.
+
+|Property|Description|Type|
+|---|---|---|
+|id|Unique identifier for the alert type.|string|
+|name|A user friendly name for the alert type.|string|
+|validate.params|Joi object validation for the parameters the executor receives.|Joi schema|
+|execute|A function to be called when executing an alert. See executor below.|Function|
+
+### Executor
+
+This is the primary function for an alert type, whenever the alert needs to execute, this function will perform the execution. It receives a variety of parameters, the following table describes the properties the executor receives.
+
+**execute(options)**
+
+|Property|Description|
+|---|---|
+|services.callCluster|Use this to do elasticsearch queries on the cluster Kibana connects to. NOTE: This currently authenticates as the Kibana internal user, this will change in a future PR.|
+|services.savedObjectsClient|Use this to manipulate saved objects. NOTE: This currently only works when security is disabled. A future PR will add support for enabled security using Elasticsearch API tokens.|
+|services.log|Use this to create server logs. (This is the same function as server.log)|
+|scheduledRunAt|The date and time the alert type was supposed to be called.|
```

Comment:
yeah I think that will come with the history / audit log phase where we could leverage that for meta functionality and such.

In regards to `meta` you're thinking wrapping those two into a `meta` object instead of being provided flat?